CMakeLists.txt

```diff
@@ -1,24 +1,19 @@
-cmake_minimum_required(VERSION 3.8)
-project(bls12-381 VERSION 1.0.0)
+cmake_minimum_required(VERSION 3.12)
 
-#set(CMAKE_CXX_STANDARD 17)
-#set(CMAKE_CXX_EXTENSIONS ON)
-#set(CMAKE_CXX_STANDARD_REQUIRED ON)
-#add_compile_options(-Wall)
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR AND NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
 
-# compile with "-march" or enable cpu features explicitly for best performance:
-#set(CMAKE_CXX_FLAGS "-madx -mbmi2")
-set(CMAKE_CXX_FLAGS "-fstack-protector-strong")
+project(bls12-381 VERSION 1.0.0)
 
-if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "Release"
-    CACHE STRING "Possible values are empty, Debug, Release, RelWithDebInfo, MinSizeRel, ..."
-    FORCE
-  )
-endif()
+file(GLOB BLS12-381_SOURCES CONFIGURE_DEPENDS src/*.cpp)
+file(GLOB BLS12-381_HEADERS CONFIGURE_DEPENDS include/bls12-381/*.hpp)
```

Comment:
I normally consider `file(GLOB` usage such as this an anti-pattern -- cmake docs even warn against it -- but it makes the `PUBLIC_HEADER` property below easier to accommodate.

Reply:
Would it make sense to use `GLOB` for headers, but list `src/` source files explicitly?

Reply:
I'm open to that; I guess I thought doing one but not the other was inconsistent?

Reply:
I could be wrong, but I had the impression that cmake did a better job with header dependency than GLOBing source files.